answers/final.py

```diff
@@ -0,0 +1,99 @@
+from collections import OrderedDict, Counter
+import logging
+
+from gensim.parsing.preprocessing import STOPWORDS
+from gensim.utils import simple_preprocess
+import networkx
+import nltk
+import spacy
+
+from answers.extraction import DATA, parse_docs
+
+parse_docs()
+
+nlp = spacy.load('en_core_web_md')
+
+def grapheize(graph, doc, attrs={}):
+    """
+    Create a graph from given piece of text. Nodes are formed from lemmatized
+    tokens. Edges are created between a lemmatized token and lemma of head of
+    the token.
+    """
+    unallowed_tags = [
+        'EX', 'HVS', 'MD', 'PDT',
+        'IN', 'DT', 'TO', 'CD',
+        'CC', '-LRB-', 'HYPH', ':'
+    ]
+    for token in doc:
+        if (token.tag_ in unallowed_tags) or (token == token.head):
+            continue
+        nodes = [token.lemma_, token.head.lemma_]
+        for node in nodes:
+            if node not in graph:
+                graph.add_node(node, token=token)
+            node = graph.node.get(node)
+            for key, value in attrs.items():
+                node.setdefault(key, []).append(value)
+        graph.add_edge(*nodes)
+
+def get_answer(question, graph, final=False):
+    q_graph = networkx.Graph()
+    q_doc = nlp(question)
+    q_type = []
+    for token in q_doc:
+        if token.tag_.startswith('W'):
```

Comment:
Nothing yet. It was supposed to be used to determine the question type and accordingly select from our top documents.